src/components/Minecraft/MinecraftCreateForm.tsx

```diff
@@ -1,72 +1,122 @@
-  const formReady = activeStep === steps.length - 1;
+  const [gameType, setGameType] = useState<HandlerGameType>(
+    'MinecraftJavaVanilla'
+  );
+  const {
+    data: setup_manifest,
+    isLoading,
+    error,
+  } = SetupInstanceManifest(gameType);
+
   const gaEventTracker = useAnalyticsEventTracker('Create Instance');
   const formikRef =
-    useRef<FormikProps<MinecraftSetupConfigPrimitiveForm>>(null);
+    useRef<FormikProps<Record<string, ConfigurableValue | null>>>(null);
 
   useEffectOnce(() => {
     gaEventTracker('Create Instance Start');
   });
 
-  const createInstance = async (value: MinecraftSetupConfigPrimitive) => {
+  useEffect(() => {
+    if (!isLoading && !error) {
+      setSetupManifest(setup_manifest);
+    }
+  }, [gameType, isLoading, setup_manifest, error]);
+
+  const [setupManifest, setSetupManifest] =
+    useState<ConfigurableManifest | null>(null);
+
+  if (setupManifest === null) return <Spinner />;
+
+  const initialValue: Record<string, ConfigurableValue | null> =
+    generateInitialValues(setupManifest['setting_sections']);
+  const validationSchema = generateValidationSchema(setupManifest);
+  const currentValidationSchema = validationSchema[activeStep];
+  const steps = [
+    'Select Game',
+    Object.keys(setupManifest['setting_sections']).map(
+      (sectionId) => setupManifest['setting_sections'][sectionId]['name']
+    ),
+  ].flat();
+  const formReady = activeStep === steps.length - 1;
+  const createInstance = async (value: ManifestValue) => {
+    console.log(JSON.stringify(value));
     await axiosWrapper<void>({
       method: 'post',
-      url: '/instance/minecraft',
-      data: value,
+      url: `/instance/create/${gameType}`,
+      headers: { 'Content-Type': 'application/json' },
+      data: JSON.stringify(value),
     });
   };
 
   async function _submitForm(
-    values: MinecraftSetupConfigPrimitiveForm,
-    actions: FormikHelpers<MinecraftSetupConfigPrimitiveForm>
+    values: Record<string, ConfigurableValue | null>,
+    actions: FormikHelpers<Record<string, ConfigurableValue | null>>
   ) {
-    const parsedValues: MinecraftSetupConfigPrimitive = {
-      ...values,
-      flavour: flavourStringToMinecraftFlavour(values.flavour),
-      cmd_args: values.cmd_args.split(' ').filter((item) => item !== ''),
-      auto_start: values.auto_start === 'true',
-      restart_on_crash: values.restart_on_crash === 'true',
-      start_on_connection: values.start_on_connection === 'true',
+    const sectionValues: Record<string, SectionManifestValue> = {};
+    for (let i = 1; i < steps.length - 1; i++) {
+      const structure = getSectionValidationStructure(values, i);
+      sectionValues[structure[1]] = structure[0];
+    }
+
+    const parsedValues: ManifestValue = {
+      auto_start: values.auto_start?.value === 'true' ? true : false,
+      restart_on_crash:
+        values.restart_on_crash?.value === 'true' ? true : false,
```

Comment:
maybe we should give this treatment (for boolean value, accept "true" string as true) to all boolean values